### ISSUE TITLE
False positive on E0602 with duplicate name

### ISSUE THREAD

**Issue by ElliottBignell** [PyCQA/pylint#2668]
Code sample: 

```
if not config.is_file():
...
    # Feature Scaling
    sc = MinMaxScaler( feature_range = ( 0, 1 ) )
...

if trainset.is_file() and scaler.is_file():

    with open( trainSetPath, 'rb' ) as input:

        dataset_train = pickle.load( input )

        with open( scalerPath, 'rb' ) as scalerInput:

            # sc = pickle.load( scalerInput ) // This was missing
...
            inputs = sc.transform( inputs ) // Runtime error here
```

### Steps to reproduce
1. Initialise variable in n-scope in earlier block
2. Exit to scope m, lower than n
3. Call method on variable with same name in lower block at scope deeper than m
3. Run
4. Stick fingers in ears and wait for bang

### Current behavior
Variable with duplicate name in scope of earlier block suppresses warning when variable with same name is initialised in lower block

### Expected behavior
Unitialised state should reflect true scope of the uninitialised variable and warn when the earlier variable name is out of scope when dereferenced later

### pylint --version output
pylint 2.2.2
astroid 2.1.0
Python 3.6.7 |Anaconda, Inc.| (default, Oct 23 2018, 19:16:44) 
[GCC 7.3.0]

### notes
First contribution. I checked for "undefined variable" and found no apparent match. Hope this is not a duplicate. If so, sorry.

I came across the problem while first trying PyLint and while refactoring some learning code, so it's fair to say I'm a beginner here, although I have used Lint and a few other tools with C++ for a long time. I THINK I have understood Python's scoping correctly, in that the earlier use of the identifier has gone out of scope at the point the warning should have been raised, so the use of the method call is legitimately an uninitialised variable. It may be fair to respond that production code needs a new identifier name which would resolve the issue, but I would demur on a variety of grounds, most saliently that the point of linting is to warn me of precisely this omission. At any rate it is AS FAR AS I CAN TELL a genuine false positive.

**Comment by PCManticore** [comment]
Hey @ElliottBignell Thanks for reporting an issue. I don't quite follow what the problem is, can you post a snippet with what error you are receiving from `pylint` and what would you expect to happen?

**Comment by PCManticore** [comment]
Closing as I cannot reproduce this. Please post a simple example that can showcase the error and we can take another look at it.
